libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -791,7 +791,50 @@ void IRGeneratorForStatements::endVisit(IndexAccess const& _indexAccess)
 		));
 	}
 	else if (baseType.category() == Type::Category::Array)
-		solUnimplementedAssert(false, "");
+	{
+		ArrayType const& arrayType = dynamic_cast<ArrayType const&>(baseType);
+		solAssert(_indexAccess.indexExpression(), "Index expression expected.");
+
+		switch (arrayType.location())
+		{
+			case DataLocation::Storage:
+			{
+				string slot = m_context.newYulVariable();
+				string offset = m_context.newYulVariable();
+
+				m_code <<
```

Comment:
Would it be more readable with whiskers?

Reply:
It would look like that:
```
				m_code << Whiskers(R"(
					let <slot>, <offset> := <indexFunc>(<dataAreaFunc>(<array>), <index>)
				)")
				("slot", slot)
				("offset", offset)
				("indexFunc", m_utils.storageArrayIndexAccessFunction(arrayType))
				("dataArearFunc", m_utils.arrayDataAreaFunction(arrayType))
				("array", m_context.variable(_indexAccess.baseExpression()))
				("index", m_context.variable(*_indexAccess.indexExpression()))
				.render();
```

I am undecided...

Reply:
I think whiskers is more readable.